sdk/python/symbolchain/symbol/Merkle.py

```diff
@@ -62,3 +75,180 @@ def calculate_next_hash(working_hash, merkle_part):
 
 	computed_root_hash = reduce(calculate_next_hash, merkle_path, leaf_hash)
 	return root_hash == computed_root_hash
+
+# endregion
+
+
+# region LeafNode / BranchNode
+
+def _get_nibble_at(path, index):
+	byte = path.path[index // 2]
+	return byte & 0xF if 1 == index % 2 else byte >> 4
+
+
+def _encode_path(path, is_leaf):
+	i = 0
+	buffer = [0x20 if is_leaf else 0]
+	if 1 == path.size % 2:
+		buffer[0] |= 0x10 | _get_nibble_at(path, 0)
+		i += 1
+
+	while i < path.size:
+		buffer.append((_get_nibble_at(path, i) << 4) + _get_nibble_at(path, i + 1))
+		i += 2
+
+	return bytes(buffer)
+
+
+class LeafNode:
+	"""Leaf node in a compact patricia tree."""
+
+	def __init__(self, path, value):
+		self.path = path
+		self.value = value
+
+	def calculate_hash(self):
+		"""Calculates node hash."""
+
+		hasher = sha3.sha3_256()
+		hasher.update(_encode_path(self.path, True))
+		hasher.update(self.value.bytes)
+		return Hash256(hasher.digest())
+
+
+class BranchNode:
+	"""Branch node in a compact patricia tree."""
+
+	def __init__(self, path, links):
+		self.path = path
+		self.links = links
```

Comment:
I assume I know this, and you know this, but how are links supposed to look like? (do we need that documented?)